tests/components/sma/test_config_flow.py

```diff
@@ -0,0 +1,170 @@
+
+    with patch("pysma.SMA.new_session", return_value=True), patch(
+        "pysma.SMA.device_info", return_value=MOCK_DEVICE
+    ), _patch_async_setup_entry() as mock_setup_entry:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            MOCK_USER_INPUT,
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == MOCK_USER_INPUT["host"]
+    assert result["data"] == MOCK_SETUP_DATA
+
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(hass, aioclient_mock):
+    """Test we handle cannot connect error."""
+    aioclient_mock.get("https://1.1.1.1/data/l10n/en-US.json", exc=aiohttp.ClientError)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        MOCK_USER_INPUT,
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_invalid_auth(hass, aioclient_mock):
+    """Test we handle invalid auth error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch("pysma.SMA.new_session", return_value=False):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            MOCK_USER_INPUT,
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_retrieve_device_info(hass, aioclient_mock):
+    """Test we handle cannot retrieve device info error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch("pysma.SMA.new_session", return_value=True), patch(
+        "pysma.SMA.read", return_value=False
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            MOCK_USER_INPUT,
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_retrieve_device_info"}
+
+
+async def test_form_unexpected_exception(hass):
+    """Test we handle unexpected exception."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with _patch_validate_input(side_effect=Exception):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            MOCK_USER_INPUT,
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_form_already_configured(hass):
+    """Test starting a flow by user when already configured."""
+    result = await hass.config_entries.flow.async_init(
```

Comment:
Add a MockConfigEntry to Home Assistant instead.

https://github.com/home-assistant/core/blob/5d57e5c06c338fc4f8b6541bbb399bc66ed63040/tests/common.py#L719